tests/Parser/parse.php

```diff
@@ -0,0 +1,87 @@
+<?php
+
+require_once dirname( __FILE__ ) . '/../../IABot/Parser/parse.php';
+require_once dirname(__FILE__) . '/../../IABot/checkIfDead.php';
+
+class parseTest extends PHPUnit_Framework_TestCase {
+
+	function __construct() {
+		parent::__construct();
+		$api = new API();
+		$this->parser = $this->getMockForAbstractClass( Parser::class, [ $api ] );
+		$this->reflection = new \ReflectionClass( get_class( $this->parser ) );
+	}
+
+	public function testFetchTemplateRegex() {
+		$placeholder = '{{{{templates}}}}';
+		$method = $this->getPrivateMethod( 'fetchTemplateRegex' );
+		$regexOptional = $this->getPrivateProperty( 'templateRegexOptional' );
+		$regexMandatory = $this->getPrivateProperty( 'templateRegexMandatory' );
+
+		$escapedTemplateArray = [ '{{cite web', 'param=foo', 'param2=bar}}' ];
+		$regex = $method->invokeArgs( $this->parser, [ $escapedTemplateArray ] );
+		$testRegex = str_replace( $placeholder, '{{cite web|param=foo|param2=bar}}', $regexOptional );
+		$this->assertEquals( $regex, $testRegex );
+
+		$escapedTemplateArray = [ '{{cite web', '', '\nparam=foo', 'param2=bar}}' ];
+		$regex = $method->invokeArgs( $this->parser, [ $escapedTemplateArray, false ] );
+		$testRegex = str_replace( $placeholder, '{{cite web||\nparam=foo|param2=bar}}', $regexMandatory );
```

Comment:
No, but I take it the test case `{{cite web||\nparam=foo|param2=bar}}` doesn't match up with what the placeholder would actually be replaced with. I simply tried to deduce what to test for. I'm going to get the actual `commObject` working and run through the code again to get some real test cases.